From e2e788bbab82a004fc9e35e17fe5ba2cd6645b0c Mon Sep 17 00:00:00 2001
From: bc Wong <bcwalrus@cloudera.com>
Date: Sun, 23 Nov 2014 21:42:37 -0800
Subject: [PATCH 0131/1173] [desktop] UTF8 decode byte strings before printing

This annoying bug affects unit tests. Since we're just printing out random byte
strings from random HTTP endpoints, there is no guarantee that we're getting
text. And we often don't, when we do file I/O with WebHDFS. This causes us to
log non-UTF8 safe stuff, which makes nosetest put non-UTF8 data into the XML
report file, which means that Jenkins can't parse it.

(cherry picked from commit 5d13ff8c695792fb9bc090cd61c8d8987afd9f54)
---
 desktop/core/src/desktop/lib/rest/resource.py |   11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/desktop/core/src/desktop/lib/rest/resource.py b/desktop/core/src/desktop/lib/rest/resource.py
index 76425e4..41d1e79 100644
--- a/desktop/core/src/desktop/lib/rest/resource.py
+++ b/desktop/core/src/desktop/lib/rest/resource.py
@@ -17,6 +17,8 @@
 import logging
 import posixpath
 
+from desktop.lib.i18n import smart_unicode
+
 LOG = logging.getLogger(__name__)
 
 
@@ -72,9 +74,12 @@ class Resource(object):
                                 allow_redirects=allow_redirects,
                                 urlencode=self._urlencode)
 
-    self._client.logger.debug(
-        "%s Got response: %s%s" %
-        (method, resp.content[:32], len(resp.content) > 32 and "..." or ""))
+    if self._client.logger.isEnabledFor(logging.DEBUG):
+      self._client.logger.debug(
+          "%s Got response: %s%s" %
+          (method,
+           smart_unicode(resp.content[:32], errors='replace'),
+           len(resp.content) > 32 and "..." or ""))
 
     return self._format_response(resp)
 
-- 
1.7.9.5

